### PR TITLE
Add migration guide

### PR DESCRIPTION
**Pull request by nvkevlu**
Add migration guide.

### Description

Add migration guide.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Review by chesterxgchen — APPROVED** [review]

File: docs/release_notes/flare_250.rst [hunk 1]
```diff
@@ -143,3 +143,116 @@ Developer Tutorial Page
...
+===============
+
+XGBoost support is enhanced in 2.5.0 to support secure training using Homomorphic Encryption (HE). The user interface is also simplified by
+setting the XGBoost parameters in the controller so all clients get the same parameters. 
+
+The main changes are:
+
+  - The xgboost params have been moved from the client configuration to server.
+  - New split_mode and secure_training parameters
+  - New :class:`SecureDataLoader<nvflare.app_opt.xgboost.histogram_based_v2.secure_data_loader.SecureDataLoader>`
```
Comment (YuanTingHsieh) [inline docs/release_notes/flare_250.rst:165]: this class is renamed: https://github.com/NVIDIA/NVFlare/blob/main/nvflare/app_opt/xgboost/histogram_based_v2/csv_data_loader.py

File: docs/release_notes/flare_250.rst [hunk 2]
```diff
@@ -143,3 +143,116 @@ Developer Tutorial Page
...
 both code and video to interactively learn how to convert and run FL in a few minutes. We also
 created a tutorial catalog to help you easily search and find the examples you are interested in.
+
+**********************************
+Migration to 2.5.0: Notes and Tips
+**********************************
+
+FLARE 2.5.0 introduces some API and behavior changes. This migration guide will help you to migrate from the previous NVFlare version
+to the current version.
+
```
Comment (YuanTingHsieh) [inline docs/release_notes/flare_250.rst:153]: should we mention that all "name" for app_opt need to be changed to the full "path" in config?

**Review by chesterxgchen — CHANGES_REQUESTED** [review]
@nvkevlu  please update

**Review by chesterxgchen — COMMENTED** [review]
@nvkevlu  @yhwen  we might need to update the document. need help from Yuhong

File: docs/release_notes/flare_250.rst [hunk 3]
```diff
@@ -143,3 +143,120 @@ Developer Tutorial Page
...
+              }
+          }
+      ]
+  }
+
+Simulator workspace structure
+=============================
+
+In 2.4.0, the server and all the clients are sharing the same simulator workspace root ``simulate_job``. Server and each client has
+its own app_XXXX job definition, but the Workspace same root folder may result in conflicting model file location.
```
[the remark this answers is not in the document]
Comment (yhwen) [reply docs/release_notes/flare_250.rst:260]: Added the tree view in docs: https://docs.google.com/document/d/1c0bBYdABiNLNuwZHNtY5OpepOAUj798NS_T2uUugCJc/edit

[the remark this answers is not in the document]
Comment (nvkevlu) [reply docs/release_notes/flare_250.rst]: added

**Review by chesterxgchen — APPROVED** [review]

**Pull request merged**